src/main/java/de/tum/in/www1/artemis/service/quiz/QuizResultService.java

```diff
@@ -0,0 +1,190 @@
+     * Note: We ignore test run participations
+     * // @formatter:on
+     *
+     * @param quizExercise the id of the QuizExercise that should be evaluated
+     * @return the newly generated results
+     */
+    private Set<Result> evaluateSubmissions(@NotNull QuizExercise quizExercise) {
+        Set<Result> createdResults = new HashSet<>();
+        List<StudentParticipation> studentParticipations = studentParticipationRepository.findAllWithEagerLegalSubmissionsAndEagerResultsByExerciseId(quizExercise.getId());
+        submittedAnswerRepository.loadQuizSubmissionsSubmittedAnswers(studentParticipations);
+
+        for (var participation : studentParticipations) {
+            if (!participation.isTestRun()) {
+                try {
+                    // reconnect so that the quiz questions are available later on (otherwise there will be a org.hibernate.LazyInitializationException)
+                    participation.setExercise(quizExercise);
+                    Set<Submission> submissions = participation.getSubmissions();
+                    QuizSubmission quizSubmission;
+                    if (submissions.isEmpty()) {
+                        log.warn("Found no submissions for participation {} (Participant {}) in quiz {}", participation.getId(), participation.getParticipant().getName(),
+                                quizExercise.getId());
+                        continue;
+                    }
+                    else if (submissions.size() > 1) {
+                        log.warn("Found multiple ({}) submissions for participation {} (Participant {}) in quiz {}, taking the one with highest id", submissions.size(),
+                                participation.getId(), participation.getParticipant().getName(), quizExercise.getId());
+                        List<Submission> submissionsList = new ArrayList<>(submissions);
+
+                        // Load submission with highest id
+                        submissionsList.sort(Comparator.comparing(Submission::getId).reversed());
+                        quizSubmission = (QuizSubmission) submissionsList.getFirst();
+                    }
+                    else {
+                        quizSubmission = (QuizSubmission) submissions.iterator().next();
+                    }
+
+                    participation.setInitializationState(InitializationState.FINISHED);
+
+                    boolean resultExisting = false;
+                    // create new result if none is existing
+                    Result result;
+                    if (participation.getResults().isEmpty()) {
+                        result = new Result().participation(participation);
+                    }
+                    else {
+                        resultExisting = true;
+                        result = participation.getResults().iterator().next();
+                    }
+                    // Only create Results once after the first evaluation
+                    if (!resultExisting) {
```

Comment:
```suggestion;
                    if (!participation.getResults().isEmpty()) {
                        continue;
                    }
                    
                    Result result = new Result().participation(participation);
```
If the result already exists, we don't do anything with it

Reply:
we only moved this code to another place, no changes needed